Sources/Utility/ProcessGroup.swift

```diff
@@ -0,0 +1,59 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Dispatch
+import Foundation
+
+public enum ProcessGroupError: Swift.Error {
+    /// The process group was cancelled and doesn't allow adding more processes.
+    case cancelled
+}
+
+/// A process group is a small warpper for collection of processes.
+/// 
+/// This class is thread safe.
+public final class ProcessGroup {
+
+    /// Array to hold the processes.
+    private var processes: [Process] = []
+
+    /// Queue to mutate internal states of the process group.
+    private let serialQueue = DispatchQueue(label: "org.swift.swiftpm.processgroup")
+
+    /// If the process group was asked to cancel all active processes.
+    private var cancelled = false
+
+    public init() {}
+
+    /// Add a process to the process group. This method will throw if the process group is terminated.
+    ///
+    /// - Parameters:
+    ///   - process: The process to add.
+    /// - Throws: ProcessGroupError
+    public func add(_ process: Process) throws {
+        return try serialQueue.sync {
+            guard !cancelled else {
+                throw ProcessGroupError.cancelled
+            }
+            self.processes.append(process)
+        }
+    }
+
+    /// Sends terminate signal to all the running processes. 
+    /// If none of the proccesses in the group are running, this method does nothing.
+    public func terminateAll() {
+        serialQueue.sync {
+            cancelled = true
+            for process in self.processes where process.isRunning {
+                process.terminate()
```

Comment:
I think we should call `interrupt()` here instead, because we want to send `SIGINT` first to give processes a chance to gracefully terminate.